x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/In.java

```diff
@@ -50,6 +52,36 @@
 import static org.elasticsearch.xpack.esql.core.type.DataType.VERSION;
 import static org.elasticsearch.xpack.esql.core.util.StringUtils.ordinal;
 
+/**
+ * The {@code IN} operator.
+ * <p>
+ *     This function has quite "unique" null handling rules around {@code null} and multivalued
+ *     fields. Let's use an example: {@code WHERE x IN (a, b, c)}. Here's the per-row
+ *     decision sequence:
+ * </p>
+ * <ol>
+ *     <li>{@code x IS NULL} => {@code null}</li>
+ *     <li>{@code a IS NULL AND b IS NULL AND c IS NULL} => {@code null}</li>
+ *     <li>{@code MV_COUNT(a) > 1 AND MV_COUNT(a) > 1 AND MV_COUNT(a) > 1} => {@code null}</li>
```

Comment:
I think you mean 
```suggestion
 *     <li>{@code MV_COUNT(a) > 1 AND MV_COUNT(b) > 1 AND MV_COUNT(c) > 1} => {@code null}</li>
```

Reply:
I also think we should be explicit about the assumption this is optimizing for, namely that `x == multi_valued_field` will always return null.